waiter/src/waiter/scheduler/kubernetes.clj

```diff
@@ -214,28 +215,34 @@
   [restart-expiry-threshold pod]
   (try
     (let [port0 (get-in pod [:spec :containers 0 :ports 0 :containerPort])
-          restart-count (get-in pod [:status :containerStatuses 0 :restartCount])
+          restart-count (get-in pod [:status :containerStatuses 0 :restartCount] 0)
           run-as-user (or (get-in pod [:metadata :labels :waiter/user])
                           ;; falling back to namespace for legacy pods missing the waiter/user label
-                          (k8s-object->namespace pod))]
+                          (k8s-object->namespace pod))
+          {:keys [phase] :as pod-status} (:status pod)
+          ;; waiter-app is the first container we register
+          primary-container-state (pc/map-vals #(or (:reason %) "reason_unknown")
```

Comment:
This would give us `{:running "reason_unknown"}`. Is that really what we want?

Reply:
Removed.